tests/unit/cmd/scan/test_prereceive.py

```diff
@@ -363,3 +364,35 @@ def test_stdin_input_deletion(
         )
         assert_invoke_ok(result)
         assert "Deletion event or nothing to scan.\n" in result.output
+
+    @patch("pygitguardian.client.GGClient.read_metadata")
+    def test_server_unavailable(
+        self,
+        read_metadata_mock: Mock,
+        tmp_path,
+        cli_fs_runner: CliRunner,
+    ):
+        """
+        GIVEN a repo on which the command is ran
+        WHEN the server is not responding (503)
+        THEN it should return 0
+        AND display an error message
+        """
+        # Set up the mock to return a 503 response
+        response = Response()
+        response.status_code = 503
+        response.detail = "Service Temporarily Unavailable"
+        read_metadata_mock.return_value = response
+
+        # setting up repo to run the command
+        repo = create_pre_receive_repo(tmp_path)
+        old_sha = repo.get_top_sha()
+        shas = [repo.create_commit() for _ in range(3)]
+        with cd(repo.path):
+            result = cli_fs_runner.invoke(
+                cli,
+                ["-v", "secret", "scan", "pre-receive"],
+                input=f"{old_sha} {shas[-1]} origin/main\n",
+            )
+        assert "Server is not responding" in result.output
+        assert_invoke_exited_with(result, 0)
```

Comment:
We usually call `assert_invoke*()` before other asserts. This makes debugging easier because if the command did not exit with the expected exit code, `assert_invoke*()` prints the command stdout and stderr.

Also there is `assert_invoke_ok()` for the case where the exit code is 0.